File: frontend/tips/3-work-with-frames/code/csharp/FramesTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;

namespace csharp
{
    [TestClass]
    public class FileUploadTest
    {
        IWebDriver _driver;

        [TestInitialize]
        public void SetUp()
        {
            _driver = new FirefoxDriver();
        }

        [TestCleanup]
        public void TearDown()
        {
            _driver?.Quit();
        }

        [TestMethod]
        public void Example1()
        {
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/nested_frames");
            _driver.SwitchTo().Frame(_driver.FindElement(By.Name("frame-top")));
            _driver.SwitchTo().Frame(_driver.FindElement(By.Name("frame-middle")));
            Assert.AreEqual("MIDDLE", _driver.FindElement(By.Id("content")).Text);
        }

        [TestMethod]
        public void Example2()
        {
            _driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/tinymce");
            _driver.SwitchTo().Frame(_driver.FindElement(By.Id("mce_0_ifr")));
            IWebElement Editor = _driver.FindElement(By.Id("tinymce"));
            string StartText = Editor.Text;
            Editor.Clear();
            Editor.SendKeys("Hello World!");
            string EndText = Editor.Text;
            Assert.AreNotEqual(EndText, StartText);

            _driver.SwitchTo().DefaultContent();
            string HeaderText = _driver.FindElement(By.CssSelector("h3")).Text;
            Assert.AreEqual("An iFrame containing the TinyMCE WYSIWYG Editor", HeaderText);
        }
    }
}

File: frontend/tips/3-work-with-frames/code/java/src/test/java/com/elemental/selenium/FramesTest.java
package com.elemental.selenium;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;

public class FramesTest {
    WebDriver driver;

    @BeforeEach
    public void beforeEach() {
        driver = new FirefoxDriver();
    }

    @AfterEach
    public void afterEach() {
        if (driver != null) {
          driver.quit();
        }
    }

    @Test
    public void exampleOne() throws Exception {
        driver.get("https://the-internet.herokuapp.com/nested_frames");
        driver.switchTo().frame("frame-top");
        driver.switchTo().frame("frame-middle");
        Assertions.assertEquals("MIDDLE", driver.findElement(By.id("content")).getText());
    }

    @Test
    public void exampleTwo() throws Exception {
        driver.get("https://the-internet.herokuapp.com/tinymce");
        driver.switchTo().frame("mce_0_ifr");
        WebElement editor = driver.findElement(By.id("tinymce"));
        String beforeText = editor.getText();
        editor.clear();
        editor.sendKeys("Hello World!");
        String afterText = editor.getText();
        Assertions.assertNotEquals(beforeText, afterText);
        driver.switchTo().defaultContent();
        Assertions.assertEquals("An iFrame containing the TinyMCE WYSIWYG Editor",
            driver.findElement(By.cssSelector("h3")).getText());
    }

}